File: examples/python/exponentiation_exponent_then_mod.py
import sys
# Ported from examples/lumen/exponentiation_exponent_then_mod.lm by scripts/port_examples.py; edit the Lumen original, not this file.
base = 7;
exp = 100;
mod = 1000000007;
iterations = 100;
print("Fast modular exponentiation benchmark");
sys.stdout.write("base = ");
print(base);
sys.stdout.write("exp  = ");
print(exp);
sys.stdout.write("mod  = ");
print(mod);
sys.stdout.write("iterations = ");
print(iterations);
print("");
print("Running mod_pow...");
result = 0;
i = 0;
while i < iterations:
    result = mod_pow(base, exp, mod);
    i = i + 1;
sys.stdout.write("Result: ");
print(result);
print("Done!");

File: examples/rust/exponentiation_exponent_then_mod.rs
// Ported from examples/lumen/exponentiation_exponent_then_mod.lm by scripts/port_examples.py; edit the Lumen original, not this file.
fn main() {
    let base = 7;
    let exp = 100;
    let mod = 1000000007;
    let iterations = 100;
    println!("Fast modular exponentiation benchmark");
    print!("base = ");
    println!("{}", base);
    print!("exp  = ");
    println!("{}", exp);
    print!("mod  = ");
    println!("{}", mod);
    print!("iterations = ");
    println!("{}", iterations);
    println!("");
    println!("Running mod_pow...");
    let mut result = 0;
    let mut i = 0;
    while i < iterations {
        result = mod_pow(base, exp, mod);
        i = i + 1;
    }
    print!("Result: ");
    println!("{}", result);
    println!("Done!");
}
